src/Datastore/RAMDataStore.cpp

```diff
@@ -0,0 +1,337 @@
+/** ----------------------------------------------------------------------------
+ * @file    RAMDataStore.h
+ * @author  Samuel Hick <samuel.hick@gmx.at>
+ * @brief   Class RAMDataStore implementation
+ * ---------------------------------------------------------------------------*/
+
+#include "Datastore/RAMDataStore.h"
+
+#include <algorithm>
+#include <ctime>
+
+#include "Types/GlobalTypes.h"
+#include "Types/Result.h"
+
+using namespace std;
+
+TResultOpt RAMDataStore::addUser(User user) {
+  // Exclusive Access to User List
+  unique_lock<shared_mutex> MyLock(mUserMutex, defer_lock);
```

Comment:
why do you defer the lock at all?

Reply:
Because the other options are "we already have the mutex" and "try lock" but i want "definitely lock even if it blocks this thread".
In order to get this behaviour i defer the locking to the next line
See https://en.cppreference.com/w/cpp/thread/lock_tag

Reply:
When omitting the second parameter the lock works as a blocking lock.
See https://en.cppreference.com/w/cpp/thread/unique_lock/unique_lock (overloading 3)

Reply:
well that simplifies it. I´ll implement that right away